iot-hub/Samples/service/AndroidSample/app/src/main/java/com/iothub/azure/microsoft/com/androidsample/MainActivity.java

```diff
@@ -0,0 +1,112 @@
+package com.iothub.azure.microsoft.com.androidsample;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.os.Handler;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import com.microsoft.azure.sdk.iot.service.devicetwin.DeviceMethod;
+import com.microsoft.azure.sdk.iot.service.devicetwin.MethodResult;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class MainActivity extends AppCompatActivity {
+
+    private final String connString = BuildConfig.ConnectionString;;
+
+    Button btnInvoke;
+
+    EditText editTxtSendMsgsIntVal;
+    DeviceMethod methodClient;
+
+    private String lastException;
+
+    private int sendMessagesInterval = 5000;
+    MethodResult result;
+
+    private final Handler handler = new Handler();
+
+    private static final Long responseTimeout = TimeUnit.SECONDS.toSeconds(200);
+    private static final Long connectTimeout = TimeUnit.SECONDS.toSeconds(5);
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState)
+    {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        btnInvoke = findViewById(R.id.btnInvoke);
+        btnInvoke.setEnabled(true);
+
+        editTxtSendMsgsIntVal = findViewById(R.id.editTxtSendMsgsIntVal);
+    }
+
+    final Runnable exceptionRunnable = new Runnable() {
+        public void run() {
+            AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
+            builder.setMessage(lastException);
+            builder.show();
+            System.out.println(lastException);
+        }
+    };
+
+    final Runnable methodResultRunnable = new Runnable() {
+        public void run() {
+            Context context = getApplicationContext();
+            CharSequence text = "Received Status=" + result.getStatus() + " Payload=" + result.getPayload();
```

Comment:
>result.getPayload() [](start = 88, length = 19)

do you need a null check here first?

Reply:
added

---
In reply to: [233960327](https://github.com/Azure-Samples/azure-iot-samples-java/pull/5#discussion_r233960327) [](ancestors = 233960327)